### ISSUE TITLE
ch5 logistic regression--Speech and Language Processing.

### ISSUE THREAD

**Issue by raychenv** [raychenv/blog#286]
Introduce an algorithm that is suited for discovering the link between features or cues and some particular outcome: logistic regression
one of the most important tools in social and natural sciences. In NLP, it is baseline supervised machine learning algorithm for classification, have close relationship with neural networks

A neural network can be viewed as a series of logistic regression classifiers
can be used to classify an obervation into one of two classes

muttinomial logistic regression

Generative and discriminative classifers
logistic regression -- discriminative classifier
naive Bayes is a generative classifier

naive Bayes
c = argmax P(d|c) P(c)
likelihood and prior

generative mode like naive Bayes makes use of this likelihood term
discriminative model directly compute P(c|d) assign high weight to document features that directly improve its ability to discriminate between possible classes

Components of a probabilistic machine learning classifier:
a training corpus of m input/output pairs (xi, yi),
1. A feature representation
2. A classification function that computes y the estimated class via p(y|x), **sigmoid**, softmax tools
3. An objective function -- cross entropy loss function
4. An algorithm for opt the obj function stochastic gradient descent algorithm

two phases:
training: weights w and b
test: a test example x, we compute p(y|x) label y=1 or 0

5.1 sigmoid function
5.2 classification with logistic regression
5.2.1 sentiment classification
5.2.2 other classification tasks and features

5.3 multinomial logistic regression
softmax
applying softmax in logistic regression
features in it

5.4 Learning in logistic regression
